src/cunumeric/random/bitgenerator.cc

```diff
@@ -0,0 +1,77 @@
+/* Copyright 2021-2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "cunumeric/random/bitgenerator.h"
+#include "cunumeric/random/bitgenerator_template.inl"
+#include "cunumeric/random/bitgenerator_util.h"
+
+#include "cunumeric/random/curand_help.h"
+
+#include "cunumeric/random/bitgenerator_curand.inl"
+
+namespace cunumeric {
+
+using namespace Legion;
+using namespace legate;
+
+struct CPUGenerator : public CURANDGenerator {
+  CPUGenerator(BitGeneratorType gentype)
+  {
+    CHECK_CURAND(::curandCreateGeneratorHost(&gen, get_curandRngType(gentype)));
+    offset             = 0;
+    type               = get_curandRngType(gentype);
+    supports_skipahead = supportsSkipAhead(type);
+    dev_buffer_size    = DEFAULT_DEV_BUFFER_SIZE;
+    dev_buffer         = (uint32_t*)::malloc(dev_buffer_size * sizeof(uint32_t));
```

Comment:
Again, if this is a temporary allocation that doesn't outlive the task, please use `create_buffer` instead.

Reply:
It does outlive the task.